collection/collection.go

```diff
@@ -384,49 +385,45 @@ func (s *localSet) handleEvent(ctx context.Context, e event.Event) error {
 				log.Debugw("adding dataset across all collections", "initID", vi.InitID, "err", err)
 			}
 		}
-
-	case event.ETAutomationDeployStart:
-		if evt, ok := e.Payload.(event.DeployEvent); ok {
+	case event.ETAutomationWorkflowStarted:
+		if evt, ok := e.Payload.(event.WorkflowStartedEvent); ok {
 			err := s.updateOneAcrossAllCollections(evt.InitID, func(vi *dsref.VersionInfo) {
-				vi.WorkflowID = evt.WorkflowID
+				vi.RunID = evt.RunID
+				vi.RunStatus = "running"
 			})
 			if err != nil {
 				log.Debugw("updating dataset across all collections", "InitID", evt.InitID, "err", err)
 			}
 		}
-	case event.ETAutomationDeployEnd:
-		if evt, ok := e.Payload.(event.DeployEvent); ok {
+	case event.ETAutomationWorkflowStopped:
+		if evt, ok := e.Payload.(event.WorkflowStoppedEvent); ok {
 			err := s.updateOneAcrossAllCollections(evt.InitID, func(vi *dsref.VersionInfo) {
-				vi.WorkflowID = evt.WorkflowID
-				if evt.Error != "" {
-					vi.WorkflowID = ""
-				}
+				vi.RunStatus = evt.Status
 			})
 			if err != nil {
```

Comment:
Okay so this is a bit odd.

1) what "kind" of run need to be represented in the collection? If a user does a save with the apply flag, should that run id and run status be captured? If so, relying on any kind of workflow or deploy event will not capture this. We should instead rely on the actually transform events. However, as of right now, we can't differentiate between a dry run or a saved run at the transform event level, which means any dry run information would be captured here as well.

2) Even if we are okay leaving out "save w/ the apply flag" type runs, we are not properly capturing any "run status" information in the collection during a deploy, because we currently disable any use of the "RunWorkflow" section, since that will cause "double runs" for dataset with transforms, since any dataset with a transform is being "applied" until we are able to save without a structure or body. Once we are able to do that, we will capture any run information in the deploy path correctly, because we will actually be using `RunWorkflow` to apply the transform.

So right now, any run information is going to be unreliable. Not sure where to settle this or what compromises we are okay with for now.

Reply:
Great articulation of the ambiguities of what "running" is. I vote you copy-paste this comment into a new issue & add a blurb about how we need more distinction for "dry runs" (apply command on the backend).

That said, this code moves us closer to something workable in the short term, so let's merge it & work out a proper fix in the new issue discussion